src/gt4py/next/program_processors/runners/dace_iterator/itir_to_tasklet.py

```diff
@@ -906,7 +1006,7 @@ def _visit_reduce(self, node: itir.FunCall):
 
             # initialize the reduction result based on type of operation
             init_value = get_reduce_identity_value(op_name.id, result_dtype)
-            init_state = self.context.body.add_state_before(self.context.state, "init")
+            init_state = self.context.body.add_state_before(self.context.state, "init", True)
```

Comment:
I know it does not belong to the PR, but why not using a library node?

Reply:
Do you mean for reduction. That's definitely something to explore, DaCe already provides it:
https://github.com/spcl/dace/blob/master/dace/libraries/standard/nodes/reduce.py